src/audio/volume/volume.c

```diff
@@ -629,22 +629,22 @@ static int volume_ctrl_get_cmd(struct comp_dev *dev,
 		return -EINVAL;
 	}
 
-	if (cdata->cmd == SOF_CTRL_CMD_VOLUME ||
-	    cdata->cmd ==  SOF_CTRL_CMD_SWITCH) {
-		comp_dbg(dev, "volume_ctrl_get_cmd(), SOF_CTRL_CMD_VOLUME / SOF_CTRL_CMD_SWITCH, cdata->comp_id = %u",
-			 cdata->comp_id);
-		for (j = 0; j < cdata->num_elems; j++) {
-			cdata->chanv[j].channel = j;
-			cdata->chanv[j].value = cd->tvolume[j];
-			comp_info(dev, "volume_ctrl_get_cmd(), channel = %u, value = %u",
-				  cdata->chanv[j].channel,
-				  cdata->chanv[j].value);
-		}
-	} else {
+	if (cdata->cmd != SOF_CTRL_CMD_VOLUME && cdata->cmd != SOF_CTRL_CMD_SWITCH) {
 		comp_err(dev, "volume_ctrl_get_cmd(): invalid cdata->cmd");
 		return -EINVAL;
 	}
 
+	for (j = 0; j < cdata->num_elems; j++) {
+		cdata->chanv[j].channel = j;
+		if (cdata->cmd == SOF_CTRL_CMD_VOLUME)
+			cdata->chanv[j].value = cd->tvolume[j];
+		else
+			cdata->chanv[j].value = !cd->muted[j];
+		comp_info(dev, "volume_ctrl_get_cmd(), channel = %u, value = %u",
+			  cdata->chanv[j].channel,
+			  cdata->chanv[j].value);
+	}
+
```

Comment:
can we move the test out of the loop? this would lead to more efficient code.... It makes no sense to test for a constant inside of a loop...

Reply:
yes, sure. I will add the original code I had with the switch and no if's insde the for loop. I was just testing to see what would help get around the CHT build issue.